agent/xdsv2/cluster_resources.go

```diff
@@ -250,8 +251,8 @@ func (pr *ProxyResources) makeEnvoyAggregateCluster(name string, protocol string
 	return clusters, nil
 }
 
-func addLocalAppHttpProtocolOptions(protocol string, c *envoy_cluster_v3.Cluster) error {
-	if !(protocol == "http2" || protocol == "grpc") {
+func addLocalAppHttpProtocolOptions(protocol pbcatalog.Protocol, c *envoy_cluster_v3.Cluster) error {
```

Comment:
My two cents is that this should come from the IR (i.e. from `ProxyState`). The idea with the IR was to be the intermediate layer of translation between consul specific resources and envoy and so we should not be passing pbcatalog* or pbmesh objects directly to envoy generation code.

Reply:
agree.  this would bleed into our abstraction.  I'll look to modify.